### PR TITLE
Create configurable client rejection message for SASL only

### PR DESCRIPTION
**Pull request by meldra**
Addresses issue #236

### CODE REVIEW

File: ircd/s_user.c [hunk 1]
```diff
@@ -491,9 +491,15 @@ register_local_user(struct Client *client_p, struct Client *source_p)
 
 	if(IsNeedSasl(aconf) && !*source_p->user->suser)
 	{
+
+		const char *sasl_only_client_message = ConfigFileEntry.sasl_only_client_message;
 		ServerStats.is_ref++;
 		sendto_one_notice(source_p, ":*** Notice -- You need to identify via SASL to use this server");
```
Comment (examknow) [inline ircd/s_user.c:497]: I'm wondering if it would also be good to show the custom message here

Comment (meldra) [reply ircd/s_user.c]: Done!

File: doc/reference.conf [hunk 2]
```diff
@@ -1148,6 +1148,11 @@ general {
 	 */
 	kline_reason = "Connection closed";
 
+        /* SASL access only client message: give users a message that
+         * informs them
+         */
```
Comment (aaronmdjones) [inline doc/reference.conf:1153]: This is indented with spaces; it must be indented with tabs to match the other lines.

File: modules/m_info.c [hunk 3]
```diff
@@ -200,6 +200,11 @@ static struct InfoStruct info_table[] = {
 		"Message to quit users with if this server is draining.",
 		INFO_STRING(&ConfigFileEntry.drain_reason),
 	},
+        {
+                "sasl_only_client_message",
+                "Message to quit users with if they require SASL authentication.",
+                INFO_STRING(&ConfigFileEntry.sasl_only_client_message),
+        },
```
Comment (aaronmdjones) [inline modules/m_info.c:207]: Same here.

**Review by aaronmdjones — CHANGES_REQUESTED** [review]

File: modules/m_info.c (hunk 3 above)
Comment (meldra) [reply modules/m_info.c:207]: Serves me right for using nano instead of an actual editor i guess.

Comment (aaronmdjones) [reply modules/m_info.c]: Hey! I've edited `doc/reference.conf` with nano several times. :P

**Review by aaronmdjones — APPROVED** [review]

**Comment by dwfreed** [comment]
Superseded

**Pull request closed without merging**